### PR TITLE
unix,fs: use uv__load_relaxed and uv__store_relaxed

### PR DESCRIPTION
**Pull request by RaisinTen**

### CODE REVIEW

**Comment by RaisinTen** [comment]
Rebased to resolve the conflicts. Could this please get a review? Thanks.

**Comment by RaisinTen** [comment]
cc @bnoordhuis

File: src/unix/fs.c [hunk 1]
```diff
@@ -965,18 +965,17 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
...
     {
-      static int copy_file_range_support = 1;
+      static int no_copy_file_range_support;
 
-      if (copy_file_range_support) {
+      if (uv__load_relaxed(&no_copy_file_range_support) == 0) {
         r = uv__fs_copy_file_range(in_fd, &off, out_fd, NULL, req->bufsml[0].len, 0);
 
-        if (r == -1 && errno == ENOSYS) {
+        if (r == -1 && ((errno == ENOSYS) || (errno == EACCES && uv__is_buggy_cephfs(in_fd)))) {
```
Comment (bnoordhuis) [inline src/unix/fs.c:973]: Can you retain the old layout? It was written that way to stay under the 80 column limit.

Comment (RaisinTen) [reply src/unix/fs.c]: @bnoordhuis Done, PTAL.

**Review by bnoordhuis — APPROVED** [review]
LGTM, thanks!

**Review by vtjnash — APPROVED** [review]

**Pull request merged**